python/chronos/src/bigdl/chronos/utils.py

```diff
@@ -91,3 +91,41 @@ def __call__(self, *args, **kwargs):
         module = importlib.import_module(module_name, package=self.pkg)
         function = getattr(module, function_name)
         return function(*args, **kwargs)
+
+
+# Refer to this guide https://www.synopsys.com/blogs/software-security/python-pickling/
+# To safely use python pickle
+class SafePickle:
+    """
+    Example:
+        >>> from bigdl.chronos.utils import SafePickle
+        >>> with open(file_path, 'wb') as file:
+        >>>     signature = SafePickle.dump(data, file)
+        >>> with open(file_path, 'rb') as file:
+        >>>     data = SafePickle.load(file, signature)
+    """
+    @classmethod
+    def dump(self, obj, file, *args, **kwargs):
+        import pickle
+        import hmac
+        import hashlib
+        pickle.dump(obj, file, *args, **kwargs)
+        try:
+            pickled_data = pickle.dumps(obj)
+            digest = hmac.new(b'shared-key', pickled_data, hashlib.sha1).hexdigest()
```

Comment:
use a constant variable for SHARED_KEY?

Reply:
> use a constant variable for SHARED_KEY?

The SHARED_KEY is only used for HMAC generation, `pickled_data` is actually authenticated using HMAC. When data is changed, `digest` is different.